### PR TITLE
 [SPARK-46274][SQL] Fix Range operator computeStats() to check long validity before converting

### PR DESCRIPTION
**Pull request by n-young-db**
### What changes were proposed in this pull request?

Range operator's `computeStats()` function unsafely casts from `BigInt` to `Long` and causes issues downstream with statistics estimation. Adds bounds checking to avoid crashing.

### Why are the changes needed?

Downstream statistics estimation will crash and fail loudly; to avoid this and help maintain clean code we should fix this.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UT

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dtenedor** [comment]
cc @cloud-fan

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/BasicStatsEstimationSuite.scala [hunk 1]
```diff
@@ -176,6 +176,25 @@ class BasicStatsEstimationSuite extends PlanTest with StatsEstimationTestBase {
         expectedStatsCboOff = rangeStats, extraConfig)
   }
 
+  test("range with invalid output") {
+    val numElements = BigInt(Long.MaxValue) - BigInt(Long.MinValue)
+    val range = Range(Long.MinValue, Long.MaxValue, 1, None)
+    val rangeStats = Statistics(
+          sizeInBytes = numElements * 8,
```
Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/BasicStatsEstimationSuite.scala:183]: Would need to fix the indentation. Has to be 2-spaced

**Review by cloud-fan — APPROVED** [review]
good catch!

**Comment by cloud-fan** [comment]
thanks, merging to master/3.5!

**Review by beliefer — COMMENTED** [review]
Late LGTM.

**Pull request closed without merging**